docs/src/components/theme-switcher.tsx

```diff
@@ -0,0 +1,55 @@
+import { MoonIcon, SunIcon } from "@radix-ui/react-icons";
+
+import { Button } from "@/components/ui/button";
+import {
+  DropdownMenu,
+  DropdownMenuContent,
+  DropdownMenuItem,
+  DropdownMenuTrigger,
+} from "@/components/ui/dropdown-menu";
+
+const setTheme = (theme: "light" | "dark" | "system") => {
+  const de = document.documentElement;
+
+  de.classList.remove("dark");
+
+  if (theme === "system") {
+    const { matches: prefersDark } = window.matchMedia(
+      "(prefers-color-scheme: dark)",
+    );
+
+    if (prefersDark) {
+      de.classList.add("dark");
+    } 
+
+    localStorage.removeItem("theme");
+  } else {
+    de.classList.add(theme);
+    localStorage.setItem("theme", theme);
+  }
+};
+
+export function ThemeSwitcher() {
+  return (
+    <DropdownMenu modal={false}>
+      <DropdownMenuTrigger asChild>
+        <Button variant="ghost" size="icon">
+          <SunIcon className="scale-100 dark:invisible" />
```

Comment:
You don't need scale-100 here, it's implied.
```suggestion
          <SunIcon className="dark:invisible" />
```